l10n_ec_withhold/wizard/wizard_create_sale_withhold.py

```diff
@@ -95,6 +118,47 @@ def validate_repeated_invoice(self):
                     )
                 )
 
+    def validate_repeated_withhold(self):
+        result = self.env["account.move"].search(
+            [
+                ("partner_id", "=", self.partner_id.id),
+                ("name", "=", f"RET {self.document_number}"),
+                ("l10n_ec_withholding_type", "=", "sale"),
+                ("l10n_latam_internal_type", "=", "withhold"),
+                ("l10n_ec_journal_type", "=", "sale"),
+            ]
+        )
+        if len(result) > 0:
+            raise UserError(_(f"Withhold {self.document_number} already exist"))
+
+    def validate_selected_invoices(self):
+        line_invoices = []
+        for withhold_line in self.withhold_line_ids:
+            line_invoices.append(withhold_line.invoice_id.name)
+        invoices = []
+        for invoice in self.invoice_ids:
+            invoices.append(invoice.name)
+
+        if set(line_invoices) != set(invoices):
```

Comment:
Una manera mas eficiente y legible seria
```suggestion
        if len(self.withhold_line_ids.invoice_id) != len(self.invoice_ids):
```